src/Engine/Combat/StrikeParams.java

```diff
@@ -67,6 +69,9 @@ public static StrikeParams buildStrikeParams(
   public double baseDamage;
   public double attackerHP;
   public double attackPower;
+  public int luckBase = 0; // Luck value if you roll 0
+  public int luckRolled = 0; // The number we plug into the RNG for luck damage
+  public int luckRolledBad = 0; // The number we plug into the RNG for negative luck damage
```

Comment:
Why are there three different combat luck variables? It doesn't look like `luckBase` is ever set to anything besides zero (so why have a separate variable), and I also don't understand the need for separate positive/negative luck variables, and calculating positive/negative separately.

Is separate positive/negative calculations a detail of how the carts do it? It seems like a single roll over a defined range is simpler and gives a more correct distribution over the range.

Reply:
`luckBase` is for providing a one-RN solution to positive-negative luck ranges, as you propose. AW1 Sonja will use it ~~and it's not uncommon in Nell rebalance ideas~~, though AW1 has some other silly subtleties to luck.
AW2+ uses a second RN for negative luck.

Two-RN makes extreme results less common; YMMV on whether that's a positive, heh.

Reply:
Two-RN reduces the impact of luck overall and turns the quoted stats into lies (though I guess AW doesn't tell the player luck percentages). Like how the two-RN FE system shifts the probability so that unlikely events happen even less often than they should.

Reply:
It does reduce the impact of luck. That's arguably a good thing.

The quoted stats aren't lies. You really can reach those damage upper and lower bounds. It's just less likely to happen than it would be under a uniform distribution.

Two-RN in FE does indeed lie to you about the true chance of a hit connecting. I'd argue that's good for gamefeel because humans are dumb, but that's not really relevant here.
AW luck damage percent doesn't directly implicate the percent *chance* of e.g. finishing off a unit - it has to line up with your normal damage vs the enemy's HP/chip damage taken in a specific way for it to matter for that at all.
AW does like to throw "%" on anything related to damage (since health is natively a percentage), and that's a little deceptive, but I wasn't planning to duplicate that issue.

Reply:
OK, that all tracks. I'd probably leave a comment to explain that the multiple variables are to support different games' different luck implementations, because it's unexpected without that context.